### DISCUSSION TITLE
Where does starship store the default modules?

### DISCUSSION THREAD

**Discussion by weeebdev** [starship/starship#1811] [Q&A]
I am using WSL2 and git status here is extremely lagging. So I wanted to change the default git which starship launches for the git modules to [custom git](https://gist.github.com/jasonboukheir/3fdab92ece236744528447a4f7f5de00)

**Comment by davidkna** [comment]
`starship` uses the `libgit2` library for getting the git status so I don't think there is a straightforward way to replace that. This will change to `git` once v1.0.0 releases with #1446 which should allow such a change.

You could try making a [custom module](https://starship.rs/config/#custom-commands) for this though.
I think someone was using a different prompt theme for providing a git status module. Another solution would be to delegate the git status module with something like the check from your link above to either a windows or linux native `starship module git_status`.

**Reply by weeebdev** [reply]
Is there a way for turning off the git status when I am in the windows filesystem?

**Reply by davidkna** [reply]
There's nothing built into starship for something like this, but you could use something else to set the `$STARSHIP_CONFIG` enviroment variable to a config file with the module disabled when you are in the windows filesystem.  Something like direnv should work for that.
